app/models/study.rb

```diff
@@ -1250,9 +1250,11 @@ def all_cells
   # return an array of all single cell names in study, will check for master list of cells or concatenate all
   # cell lists from individual expression matrices
   def all_cells_array
-    arrays = DataArray.where(study_id: self.id, linear_data_type: 'Study', linear_data_id: self.id, name: 'All Cells').order_by(&:array_index)
-    if arrays.any?
-      arrays.pluck(:values).reduce(&:+)
+    if !!study.metadata_file&.parsed? # nil-safed via &, !! converts to boolean: nil|false == false
```

Comment:
I think the `!!` is confusing here.  It's not necessary because ruby will treat already nil as false for `if` purposes.  Typically you only need the `!!` if you're printing out values or passing them to another function that might differentiate between actual `false` and false-y values.

Reply:
>  the !! wouldn't help either, because it would make it true

That is not correct - `!!nil` equates to `false`, as seen here:

```
irb(main):001:0> !!nil
=> false
irb(main):002:0> !nil
=> true
irb(main):003:0> !!false
=> false
irb(main):004:0> !!true
=> true
```

If it didn't, then `nil` being treated like `false` wouldn't be the expected behaviour.  That being said, I take the point about the rest of it - since this is the default behavior for `nil`, the double-bang isn't necessary here.

Reply:
2a29c4b